docs/cromwell_features/HogFactors.md

```diff
@@ -0,0 +1,163 @@
+
+Thus:
+
+- Every workflow is assigned to exactly one hog group.
+- Every hog group may have many workflows assigned to it.
+
+The limits discussed below happen at the granularity of the hog group.
+
+### Hog Factor
+
+The hog factor is an integer greater than or equal to 1. It represents a trade-off between: 
+
+- Maximizing the resources of Cromwell and allowing jobs to be processed as fast as possible.
+- Remaining responsive to small requests from other groups - even if a few greedy groups have enough work that they 
+could otherwise max out Cromwell's limits by themselves.
+
+Here are a few mental models which might be helpful to thinking about the hog factor:
+
+- A hog factor of 2 means that "2 greedy users would be able to hog the entire resources of Cromwell" 
+- A hog factor of 100 means "any 1 group is only ever allowed to use 1/100th of the resources of the total Cromwell server"
+
+### Hog Limit
+
+Hog Limits are not set directly, but are values that Cromwell calculates internally. 
+They are the evaluated upper bounds on how much of a given resource type could be given, in total, to workflows within 
+a single greedy group (for example in the sentence "this Cromwell has calculated a hog limit of 2 jobs per group"). 
+
+## Configuration
+
+Cromwell accepts two configuration values for hog factors in the `hog-safety` stanza of the configuration file:
+```conf
+  hog-safety {
+    workflow-option = "hogGroup"
+    hog-factor = 1
+  }
+```
+
+### Setting a hog-factor
+
+The hog factor option sets the integer described in the [Hog Factor](#hog-factor) section above.
+The default value is `1` (which is equivalent to not limiting by hog group). 
+
+### Assignment of hog groups
+
+Within the configuration file, you can specify the workflow option that will determine the hog group:
+
+- The default is `hogGroup`
+- Therefore, if a workflow arrives with the following workflow options file, Cromwell will assign `hogGroupA` as the
+workflow's hog group:
+```json
+{
+  "hogGroup": "hogGroupA"
+}
+```
+
+- Any `String` field can be chosen from workflow options:
+    + You can come up with a new field and set it specifically for assigning hog groups.
+    + You can choose a field that is already being used for other reasons
+- If a workflow is submitted which does have the designated field in its workflow options, the workflow ID is used as
+the hog group. 
+
+## Effects
+
+### Job Execution
+
+#### Reserving Space
+
+- Cromwell allows administrators to designate an overall maximum concurrent job limit per backend. 
+- Within that limit, hog factors now allow us to limit the maximum concurrent jobs started per hog group.
+    + This means that new workflows can immediately run even if big workflows are queued up.
+
+#### Round robin allocation
+
+Rather than assigning job tokens on a strict first-come first served basis, Cromwell now assigns in a round-robin
```

Comment:
What is a job token? It seems to be related to queuing and what order that Cromwell starts jobs. 
And what is round robin allocation in this context?